bug-pattern/src/main/java/jp/skypencil/findbugs/slf4j/IllegalPassedClassDetector.java

```diff
@@ -7,16 +7,24 @@
 import static org.apache.bcel.Const.LDC_W;
 import static org.apache.bcel.Repository.lookupClass;
 
+import java.util.Deque;
+import java.util.LinkedList;
+
+import org.apache.bcel.classfile.ClassFormatException;
+import org.apache.bcel.classfile.Constant;
+import org.apache.bcel.generic.Type;
+
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Objects;
+
 import edu.umd.cs.findbugs.BugInstance;
 import edu.umd.cs.findbugs.BugReporter;
 import edu.umd.cs.findbugs.OpcodeStack.CustomUserValue;
 import edu.umd.cs.findbugs.OpcodeStack.Item;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
 import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
-import java.util.Deque;
-import java.util.LinkedList;
-import org.apache.bcel.classfile.Constant;
-import org.apache.bcel.generic.Type;
+import edu.umd.cs.findbugs.classfile.DescriptorFactory;
+import edu.umd.cs.findbugs.internalAnnotations.SlashedClassName;
 
 @CustomUserValue
 public class IllegalPassedClassDetector extends OpcodeStackDetector {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 7 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)